controllers/common/common_status_update.go

```diff
@@ -0,0 +1,403 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	policiesv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
+)
+
+// RootStatusUpdate updates the root policy status with bound decisions, placements, and cluster status.
+func RootStatusUpdate(ctx context.Context, c client.Client, rootPolicy *policiesv1.Policy) (DecisionSet, error) {
+	log.Info("Updating the root policy status")
```

Comment:
This needs to be put after the `if reflect.DeepEqual(rootPolicy.Status.Status, cpcs) &&` if statement in case you return without doing a status update.

Also, please include the policy name and namespace a key/value pairs in the log message.